### PR TITLE
feat(plugin): add legend gradient field component

### PR DESCRIPTION
**Pull request by nina992**
# Overview
Add the field component (legend gradient)
## How I tested
test project (temp link ) :https://test.reearth.dev/edit/01gw4bkn0p6qq9egkvv4kx6nvf

## Screenshot
image
image

## Design 
image

### CODE REVIEW

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { ReearthApi } from "@web/extensions/sidebar/types";
 export const generalFieldName = {
   idealZoom: "カメラ",
   legend: "凡例",
+  legendGradient: "Legend Gradient",
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts:7]: 凡例（グラデーション）

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
+//import { getColorsGradients } from "@web/extensions/sidebar/utils";
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts:1]: REmove please

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+          ...lg,
+          max,
+        };
+        onUpdate(newLegendGradient);
+        return newLegendGradient;
+      });
+    },
+    [onUpdate],
+  );
+  const generateDisplayValue = useCallback(
```
[the remark this answers is not in the document]
Comment (KaWaite) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts:112]: Ah I guess its a bit different. Any reason they are different?

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts (hunk 2 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts:1]: done

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts (hunk 1 above)
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts:7]: done

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts:112]: Here I don't have field ,I only have min and max values so I created like an object within it the key is the step and the value is the color

[the remark this answers is not in the document]
Comment (nina992) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts:112]: The generateConditions function generates an array of conditions and colors, while the generateDisplayValue function generates an object that maps each value to a color.

[the remark this answers is not in the document]
Comment (KaWaite) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/LegendGradient/hooks.ts:111]: Gotcha. so they are completely different :P ok!

**Review by KaWaite — APPROVED** [review]

**Pull request merged**